frontend/app/components/Map.vue

```diff
@@ -0,0 +1,94 @@
+<script setup lang="ts">
+import { defineProps, onMounted, onUnmounted, ref } from 'vue'
+
+interface Element {
+  x: number
+  y: number
+  content: string
+}
+
+defineProps<{
+  elements: Element[]
+}>()
+
+const mapPosition = ref(0)
+const isDragging = ref(false)
+const startX = ref(0)
+const startPosition = ref(0)
+
+function handleKeyPress(event: KeyboardEvent) {
+  if (event.key === 'd' || event.key === 'ArrowRight') {
```

Comment:
This page is one of the most important ones, make sure using arrows is smooth. Store velocity, request animation frames, add css duration, anything to make it less 'laggy" and annoying